CMakeLists.txt

```
@@ -237,7 +237,7 @@ endif()
 
 ######################################################
 
-if (BTCPP_UNIT_TESTS)
+if (BTCPP_UNIT_TESTS AND BUILD_TESTING)
```

To maintain the current behavior of "tests by default", I'd suggest using `include(CTest)` to set the default value of `BUILD_TESTING`. If you do that, you won't need to change your CI workflows.

```suggestion
include(CTest)
if (BTCPP_UNIT_TESTS AND BUILD_TESTING)
```

---

Relevant docs: https://cmake.org/cmake/help/latest/module/CTest.html

---

good catch!  I'll commit that suggestion.

I am still tempted to leave in the `-DBUILD_TESTING` in the various CI configs since they explicitly depend on the tests being enabled; leaving it in would protect against future changes that move away from "tests by default".  I'll leave that decision up to @facontidavide though, happy to revert if desired.